### ISSUE TITLE
✅ 🐞 Bug Report: ACF Dynamic Background Image does not load properly on the first Loop Item in a Loop Grid - [ED-12415]

### ISSUE THREAD

**Issue by portrayaloflife** [elementor/elementor#22368]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

On several sites where we're using the new loop feature and loop grid we're seeing an issue where the very first loop item does not consistently load the background image that it's pulling from the post type. 

### Steps to reproduce

Create a few post type items, use the feature image to pull for the background image of the loop item. Add loop grid to page. 

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: Flywheel/5.1.0
	MySQL version: (Google) v5.7.40
	PHP Version: 7.4.33.2
	PHP Memory Limit: 128M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 300M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: There are some writing permissions issues with the following directories/files:
		 - WordPress root directory
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 256M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Version: 1.2
	Child Theme: Yes
	Parent Theme Name: Astra
	Parent Theme Version: 4.1.3
	Parent Theme Author: Brainstorm Force

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields
		Version: 6.1.6
		Author: WP Engine

	Apollo Social Cards
		Version: 4.0
		Author: Apollo Artistry

	Astra Pro
		Version: 4.1.4
		Author: Brainstorm Force

	Custom Post Type UI
		Version: 1.13.5
		Author: WebDevStudios

	Elementor
		Version: 3.13.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.13.0
		Author: Elementor.com

	JetElements For Elementor
		Version: 2.6.9
		Author: Crocoblock

	Redirection
		Version: 5.3.10
		Author: John Godley

	Ultimate Addons for Elementor
		Version: 1.36.15
		Author: Brainstorm Force

	User Role Editor Pro
		Version: 4.63.5
		Author: Vladimir Garagulia

	Yoast SEO
		Version: 20.6
		Author: Team Yoast


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==


== Elementor Experiments ==
	Optimized DOM Output: Inactive by default
	Improved Asset Loading: Inactive by default
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Inactive
	Additional Custom Breakpoints: Inactive
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Editor Top Bar: Inactive by default
	Landing Pages: Inactive
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Page Transitions: Inactive
	Notes: Inactive
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Inactive
	Menu: Inactive by default


== Log ==
	
Log: showing 20 of 352023-05-04 17:31:07 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.1',
  'to' => '3.12.2',
)]
2023-05-04 17:31:16 [info] elementor::elementor_updater Started 
2023-05-04 17:31:16 [info] Elementor/Upgrades - _on_each_version Start  
2023-05-04 17:31:16 [info] Elementor/Upgrades - _on_each_version Finished 
2023-05-04 17:31:16 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.1',
  'to' => '3.12.2',
)]
2023-05-04 17:31:36 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.2',
  'to' => '3.12.3',
)]
2023-05-04 17:31:56 [info] elementor-pro::elementor_pro_updater Started 
2023-05-04 17:31:56 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-05-04 17:31:56 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-05-04 17:31:56 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.2',
  'to' => '3.12.3',
)]
2023-05-08 22:56:28 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.2',
  'to' => '3.13.0',
)]
2023-05-08 22:57:37 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.3',
  'to' => '3.13.0',
)]
2023-05-09 02:46:46 [info] elementor::elementor_updater Started 
2023-05-09 02:46:46 [info] Elementor/Upgrades - _on_each_version Start  
2023-05-09 02:46:46 [info] Elementor/Upgrades - _on_each_version Finished 
2023-05-09 02:46:46 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.2',
  'to' => '3.13.0',
)]
2023-05-09 02:46:47 [info] elementor-pro::elementor_pro_updater Started 
2023-05-09 02:46:47 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-05-09 02:46:48 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-05-09 02:46:48 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.3',
  'to' => '3.13.0',
)]

PHP: showing 6 of 6PHP: 2023-05-05 04:10:35 [error X 1][/www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::295] Uncaught Error: Call to a member function update_meta() on null in /www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:295
Stack trace:
#0 /www/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(168): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(5922, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /www/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /wordpress/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /wordpress/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#5 /wordpress/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#6 /wordpress/wp-admin/admin-ajax.php(188): do_ac [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-05-09 02:55:20 [notice X 8][/www/wp-content/plugins/elementor/app/modules/import-export/module.php::696] Trying to get property 'label' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-05-09 02:55:21 [notice X 7][/www/wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-05-09 02:58:42 [notice X 8][/www/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-text.php::33] Undefined offset: 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-05-09 03:07:35 [notice X 9][/www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined index: condition_type [array (
  'trace' => '
#0: /www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: /www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: /www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: /www/wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-05-09 03:49:44 [notice X 2][/www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined index: condition_type [array (
  'trace' => '
#0: /www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /www/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(84): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: /www/wp-content/plugins/elementor/data/base/endpoint.php(166): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> update_item()
#3: /www/wp-content/plugins/elementor/data/base/endpoint.php(307): Elementor\Data\Base\Endpoint -> base_callback()
#4: /wordpress/wp-includes/rest-api/class-wp-rest-server.php(1181): Elementor\Data\Base\Endpoint -> Elementor\Data\Base\{closure}()
',
)]

JS: showing 1 of 1JS: 2023-05-09 03:21:43 [error X 12][https://REDACTED.mysites.io/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.13.0:2:8297] Cannot read properties of undefined (reading 'attributes') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified
	JetElements For Elementor: Compatibility not specified
	Ultimate Addons for Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	

				
````

</details>

**Comment by nicholaszein** [comment]
Hello @portrayaloflife! 

Thank you for submitting your issue! 🙏 

Please clarify if you are using the Featured Image (native) field, or if you are using an ACF image field. If possible share the field settings here. In your title you mentioned ACF, but in your description, you mentioned the Feature Image.

It would also help if you share a screen recording of the issue happening on a live website, or share the link of the website so we can take a look.

I'll wait for your feedback before continuing.

**Comment by portrayaloflife** [comment]
Seeing it with both Feature image backgrounds as well as ACF image backgrounds on loop grids. I'll spin up a sandbox and send over vids.

**Comment by mateitudor** [comment]
I can confirm, having the same exact issue.

**Comment by enuzzo** [comment]
I can confirm i have the same exact issue ... using Custom Posts created with Jet Engine with Crocoblock: first item in the loop does not load the background

**Comment by portrayaloflife** [comment]
@nicholaszein any update on this?

**Comment by mateitudor** [comment]
More details: I can replicate it 100% of the time with the default featured image as well, not just ACF or JetEngine, so it's definitely an Elementor bug.
You can turn off all the markup / lazy loading features and it still happens.

**Comment by ofmarconi** [comment]
#metoo

**Comment by ofmarconi** [comment]
@nicholaszein a detail that I noticed is that when we click on Regenerate CSS & Data this error always happens on the first load of the page. Always with me.

This is really ugly, I'm delivering big projects and my client will think Elementor is a joke with such a horrible error like this.

**Comment by callmehuyv** [comment]
That would be awesome if Elementor team can priority to fix this bug.

**Comment by mateitudor** [comment]
Still happening in 3.16.0-dev3 / Pro beta3

**Comment by ph-webetc** [comment]
Also seeing this issue - always the first instance of any Elementor Loop Grid, including where alternate templates are applied (we end up with missing first image for each template applied). It's becoming a little tedious now.

**Comment by ofmarconi** [comment]
shameful

**Comment by GringoWhoowhere** [comment]
Please fix

**Comment by cha5ls** [comment]
Experiencing the same issue using featured image in a loop on a popup.

**Comment by unav4ila8le** [comment]
I am still having the same issue. Last time was August, now today as well.
For those dealing with background colors, there is a simple solution: just manually input the custom field key in, it shoudl work.

**Comment by zakjechips** [comment]
Same here... on 3 different domains, just a normal loop grid, i do have ACF but it's not used in this loop.

**Comment by GringoWhoowhere** [comment]
https://github.com/elementor/elementor/issues/21489 - Similar issue here that was closed 3 weeks ago.

Pls fix.

**Comment by tristan-weil** [comment]
I can confirm I have the same issue with "Featured Image" with a Loop Grid or a Carousel (even worse as I sometimes have multiple blank background)

**Comment by Qadir9813** [comment]
Its still occuring in 3.17 pro.

**Comment by Qadir9813** [comment]
Elementor, with over 400 team members, has had this issue for six months. In that time, a new baby is likely to be born.
Please fix now or we have to wait for another baby to born?

**Comment by watels** [comment]
same issue for me...

**Comment by Richardevertse** [comment]
my featured image in loop grid is not displayed correct :-( Is this being worked on?

**Comment by brent-e** [comment]
Elementor, you really, REALLY, need to fix this soon. I'm encountering this repeatedly on both production and staging sites.

![image](https://github.com/elementor/elementor/assets/45741358/28b78c5f-f892-4309-976a-3693de108ae1)

**Comment by dgowell** [comment]
Same issue here too. I don't use ACF but i have jet engine filter but it's not used here. <img width="1192" alt="Screenshot 2023-12-04 at 17 28 51" src="https://github.com/elementor/elementor/assets/61519004/257b16ef-a5be-4d0b-88fd-8ae605b49b25">

**Comment by ofmarconi** [comment]
@elementorbot shame

**Comment by Rayken** [comment]
This is still a problem and it occurs regardless of ACF. Using a loop with a featured image removes the first image of the first post. Background-image is nowhere to be seen. Also, it appears to be on *every* first post. So if it's a WP post in a news grid or a custom post type, when the bug occurs they both get affected - at least in my testing. 

Shouldn't be months between responses, Elementor, step it up.

@danielkatz @StyleShit

**Comment by mikriz** [comment]
same issue wp 6.4.2 / elememtor 3.18.1 Anyone have any luck???

<img width="1118" alt="Screenshot 2023-12-06 at 4 40 29 PM" src="https://github.com/elementor/elementor/assets/21971636/c69a1d18-5320-4205-9f1d-3e7c876e4510">

**Comment by francescokant** [comment]
Same problem also present on my projects. In my case I have a carousel loop in which I have a container with a dynamic post featured image.

Even if you remove lazy loading, the images don't load instantly but you need to make a slide to make them appear.

![Screenshot 2023-12-14 alle 12 41 48](https://github.com/elementor/elementor/assets/143516790/250de4c0-3608-4356-94bc-be48e25f017d)

**Comment by schmidtdesign** [comment]
This bug is still alive! It's not consistent for me. It's so random. The first card will be fine and then one day the background image will disappear when arriving at the page. Click refresh and the image is back, but visitors are not going to refresh a page that just loaded. 
![Screen Shot 2023-12-19 at 9 34 13 AM](https://github.com/elementor/elementor/assets/39386976/9f4dc8c8-3d86-438a-ba8e-a593da78f394)

**Comment by mateitudor** [comment]
Maybe the devs "responsible" for this are in the iof

**Comment by callmehuyv** [comment]
Hey guys, this issue happened for a years, as a developer, do not wait them to fix it.
The workaround is always insert a image (hidden by CSS), then write Javascript to check and update the first image if it is missing.

**Comment by francescokant** [comment]
> Hey guys, this issue happened for a years, as a developer, do not wait them to fix it. The workaround is always insert a image (hidden by CSS), then write Javascript to check and update the first image if it is missing.

Finally some good soul who responds, thank you very much!
In my case I don't have an image element but the image is loaded via background. Could it still work?

**Comment by MarkoSmiles** [comment]
I'm having this same issue as well. Come on Elementor.. When can we expect this to be fixed??

**Comment by pcodina** [comment]
I have the same problem, with ACF the background images do not load. Please Elementor fix it now.



> Hey guys, this issue happened for a years, as a developer, do not wait them to fix it. The workaround is always insert a image (hidden by CSS), then write Javascript to check and update the first image if it is missing.

[callmehuyv](https://github.com/callmehuyv) Thank you so much. Can you explain how to do it?

**Comment by sixgunnar** [comment]
I've had to solve it by replacing the background images by regular images. If you set the width (100%) and height (px) under Style you can enable the option Object fit cover.

**Comment by georgeantonios** [comment]
Issue persists Elementor Pro 3.18 both with Featured Image & ACF Field. Any images hidden from pagination are not loaded.

**Comment by georgeantonios** [comment]
Also, same thing happens to videos via ACF's WYSIWYG field.

**Comment by rebofel** [comment]
Happy 2024 to all this beautiful group. 
Another buggy web here

**Comment by pcodina** [comment]
> I have the same problem, with ACF the background images do not load. Please Elementor fix it now.
> 
> > Hey guys, this issue happened for a years, as a developer, do not wait them to fix it. The workaround is always insert a image (hidden by CSS), then write Javascript to check and update the first image if it is missing.
> 
> [callmehuyv](https://github.com/callmehuyv) Thank you so much. Can you explain how to do it?

Hi [callmehuyv]

I use ACF to create a post type that I will use for a slider, among the fields there is one for the image. 

With the elementor theme builder I create a loop element to make this slider with 2 columns, in the first column is the text and in the second column is the image as background. All dynamic elements from the post type.

After that, on the home page, I have an elementor loop carousel widget where I query the items in the post type. 

Sometimes when posting, the background images are not shown, if I refresh the browser they are visible but it is not a stable system.

**Comment by mateitudor** [comment]
> Hey guys, this issue happened for a years, as a developer, do not wait them to fix it. The workaround is always insert a image (hidden by CSS), then write Javascript to check and update the first image if it is missing.

Very elegant fix, let's bloat our websites with requests we don't need.

**Comment by martkarls** [comment]
Any plans on fixing this annoying bug?

**Comment by oliver-hekmatara** [comment]
You mention in your latest release notes that this issue was resolved but I'm still seeing it. Image of the first item in a loop randomly fails to load. I don't understand how this can be so hard to fix..

**Comment by MarkoSmiles** [comment]
This issue is not resolved

**Comment by zanga7** [comment]
Same issue for me :-(

**Comment by ERion42** [comment]
Hi there- same issue (or at least a similar one) here. For me all the images in my loop carousel load on the first go-round but after that I start seeing the page loaded without an image. I've tried reconfiguring the images so they're all the same size JUST IN CASE but that hasn't worked out. Has there been any progress on this bug?
![image](https://github.com/elementor/elementor/assets/17477215/4c95fcd6-c9a9-42fb-a52f-941520010411)

**Comment by nicholaszein** [comment]
Hey everyone!

I know I'm little late, but I'm just passing to let you know that...

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.19.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
